ecobee/beam_batch.py

```diff
@@ -0,0 +1,144 @@
+import argparse
+import logging
+import os
+from collections import namedtuple
+from datetime import datetime
+from time import time
+
+import apache_beam as beam
+from apache_beam.io import ReadFromText
+from apache_beam.io import WriteToText
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+
+DataPoint = namedtuple("DataPoint", ["time", "action"])
+Sequence = namedtuple("Sequence", ["seq_len", "datapoints"])
+
+timestamp_prefix = "TS:"
+timestamp_postfix = "GMT "
+action_prefix = "Action:"
+date_fmt = "%Y-%m-%d %H:%M:%S %z"
+
+
+def to_data_point(item):
+    """
+    item =  TS:2018-12-03 13:29:15 +0000 GMT, Action:F
+    and converts it to DataPoint (2018-12-03 13:29:15 +0000, F) with first value as time (prefix and postfix removed)
+    and second value as just the action (prefix removed).
+    """
+    entry = item[: len(item) - 1].split(",") if item.endswith("\n") else item.split(",")
+    t, action = (
+        entry[0][len(timestamp_prefix) : -len(timestamp_postfix)],
+        entry[1][len(action_prefix) + 1 :],
+    )
+    return DataPoint(t, action)
+
+
+def process(words, time_period, top_n_elems):
+    """
+    :param words: list of DataPoints, in increasing order of time.
+    :param time_period if two events are apart by this value of time, they are considered in separate sequence
+    :param top_n_elems: the number of top sequences to return
+    :return: list of sequences
+    """
+    all_sequences = []
+    prev_item = words[0]
+    cur_sequence_actions = [prev_item]
+    cur_seq_len = 0
+    for item in words[1:]:
```

Comment:
use `len()` and `range()` here instead of using list slices. List slices are harder to read

Reply:
I find slices easier, leaving as is for now